plugin/CactbotEventSource/FateWatcher.cs

```diff
@@ -124,35 +125,116 @@ public class FateWatcher {
 
     public unsafe void ProcessMessage(byte* buffer, byte[] message) {
       int a = *((int*)&buffer[Category_Offset]);
-      switch (a) {
-        // Fate Start: 0x935
-        // param1: fateID
-        // param2: unknown
-        case 0x935: {
-          AddFate(*(int*)&buffer[Param1_Offset]);
-          break;
-        };
-
-        // Fate End: 0x936
-        // param1: fateID
-        case 0x936: {
-          RemoveFate(*(int*)&buffer[Param1_Offset]);
-          break;
-        };
-
-        // Fate Progress: 0x93E
-        // param1: fateID
-        // param2: progress (0-100)
-        case 0x93E: {
-          int param1 = *(int*)&buffer[Param1_Offset];
-          int param2 = *(int*)&buffer[Param2_Offset];
-          if (!fates.ContainsKey(param1)) {
-            AddFate(param1);
+      if (language_ == "ko") {
+        switch (a) {
+          //
+          // for FFXIV KO version: 5.11
+          //
+          // Latest KO version can be found at:
+          // https://www.ff14.co.kr/news/notice?category=3
+          //
+          // Fate Start: 0x74
+          // param1: fateID
+          // param2: unknown
+          case 0x74: {
```

Comment:
Would it be possible to just have a small lookup table of these offsets by language, rather than repeating all the switch statement logic?

In other words, something like if (offset == offsets[lang].add) { } etc

Reply:
Yes that is much simpler.